chalk-solve/src/clauses/builtin_traits/sized.rs

```diff
@@ -43,3 +34,53 @@ pub fn add_sized_program_clauses<I: Interner>(
     };
     builder.push_clause(trait_ref.clone(), iter::once(last_field_sized_goal));
 }
+
+fn handle_tuple<I: Interner>(
+    db: &dyn RustIrDatabase<I>,
+    builder: &mut ClauseBuilder<'_, I>,
+    trait_ref: &TraitRef<I>,
+    arity: usize,
+    substitution: &Substitution<I>,
+) {
+    // Empty tuples are always Sized
+    if arity == 0 {
+        builder.push_fact(trait_ref.clone());
+        return;
+    }
+
+    let interner = db.interner();
+
+    // To check if a tuple is Sized, we only have to look at its last element.
+    // This is because the WF checks for tuples require that all the other elements must be Sized.
+    let last_elem_sized_goal = TraitRef {
+        trait_id: trait_ref.trait_id,
+        substitution: Substitution::from1(interner, substitution.iter(interner).last().unwrap()),
+    };
+    builder.push_clause(trait_ref.clone(), iter::once(last_elem_sized_goal));
```

Comment:
this part is the same for both `handle`s, moreover I feel like this pattern will repeat in `Copy` impls. The pattern I mean is "to prove `trait_ref`, you need to prove the same trait for types `T1..Tn`", which is actually how rustc handles these things. Maybe it's better to encapsulate it in a helper function, something like:
```
fn needs_impl_for_tys(builder: &dyn RustIrDatabase<I>, trait_ref: &TraitRef<I>, tys: impl Iterator<Item = Ty<I>>) {
 ...
}
```

Reply:
Where do you think this helper function should be defined?

Reply:
In `builtin_traits.rs`, I guess?